sound/soc/sof/topology.c

```diff
@@ -492,6 +492,16 @@ static int get_token_u16(void *elem, void *object, u32 offset, u32 size)
 	return 0;
 }
 
+static int get_token_uuid(void *elem, void *object, u32 offset, u32 size)
+{
+	struct snd_soc_tplg_vendor_uuid_elem *velem = elem;
+	u8 *dst = (u8 *)object + offset;
```

Comment:
I never realized we have a integer-to-pointer cast to extract tokens, is this really necessary @ranj063?

Reply:
@plbossart this looks like a `void *` to `u8 *` cast to me, which is done to make pointer arithmetics valid not with GCC only.

Reply:
yes, @plbossart all elem/object/velem/dst above are pointers, only the offset is integer, did you mean the 'offset' to dst? This is common usage of moving a pointer, no?

Reply:
I need new reading glasses... No issues here.